src/InfiniteList/InfiniteList.tsx

```diff
@@ -157,125 +162,111 @@ export default class InfiniteList extends React.Component {
 
 		if (this.state.actionTriggered) return;
 
-		let atBottom = this.isElementAtBottom(target);
+		const atBottom = this.isElementAtBottom(target);
 
 		if (atBottom && this.props.hasMore && this.props.dataLength) {
 			this.setState(ps => ({ ...ps, actionTriggered: true, showLoader: true }));
-			this.props.next();
+			this.props.next && this.props.next();
 		}
 		this.setState(ps => ({ ...ps, lastScrollTop: target.scrollTop }));
 	};
 
-	throttle = (fn, threshhold, scope) => {
-		threshhold || (threshhold = 250);
-		let last, deferTimer;
-		return function() {
-			const context = scope || this;
-
-			const now = +new Date(),
-				args = arguments;
-			if (last && now < last + threshhold) {
-				clearTimeout(deferTimer);
-				deferTimer = setTimeout(function() {
-					last = now;
-					fn.apply(context, args);
-				}, threshhold);
-			} else {
-				last = now;
-				fn.apply(context, args);
-			}
+	private throttledOnScrollListener = throttle(this.onScrollListener, 150);
+
+	static getDerivedStateFromProps(props, state) {
+		if (props.dataLength === state.dataLength) return null;
+
+		return {
+			dataLength: props.dataLength,
+			showLoader: false,
+			actionTriggered: false,
+			pullToRefreshThresholdBreached: false
 		};
-	};
+	}
+
+	componentDidMount() {
+		this._scrollableNode = this.getScrollableTarget();
+		this.el = this.props.height ? this._infScroll : this._scrollableNode || window;
+		this.el.addEventListener("scroll", this.throttledOnScrollListener);
+
+		if (this.props.pullDownToRefresh) {
+			this.el.addEventListener("touchstart", this.onStart);
+			this.el.addEventListener("touchmove", this.onMove);
+			this.el.addEventListener("touchend", this.onEnd);
+
+			this.el.addEventListener("mousedown", this.onStart);
+			this.el.addEventListener("mousemove", this.onMove);
+			this.el.addEventListener("mouseup", this.onEnd);
+
+			this.maxPullDownDistance = this._pullDown.firstChild.getBoundingClientRect().height;
+			this.forceUpdate();
+
+			if (typeof this.props.refreshFunction !== "function") {
+				throw new Error(
+					`Mandatory prop "refreshFunction" missing.
+					Pull Down To Refresh functionality will not work
+					as expected.`
+				);
+			}
+		}
+	}
+
+	componentWillUnmount() {
+		this.el.removeEventListener("scroll", this.throttledOnScrollListener);
+
+		if (this.props.pullDownToRefresh) {
+			this.el.removeEventListener("touchstart", this.onStart);
+			this.el.removeEventListener("touchmove", this.onMove);
+			this.el.removeEventListener("touchend", this.onEnd);
+
+			this.el.removeEventListener("mousedown", this.onStart);
+			this.el.removeEventListener("mousemove", this.onMove);
+			this.el.removeEventListener("mouseup", this.onEnd);
+		}
+	}
 
 	render() {
 		const style = {
 			height: this.props.height || "auto"
 		};
-		const hasChildren = this.props.hasChildren || !!(this.props.children && this.props.children.length);
+		const hasChildren = this.props.hasChildren || !!this.props.children;
```

Comment:
This is not equivalent to previous version. In new version empty array will evaluate to True